core/unit_test/CMakeLists.txt

```diff
@@ -1284,6 +1284,29 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
   ARGS "one 2 THREE"
 )
 
+KOKKOS_ADD_EXECUTABLE(
+  CoreUnitTest_CMakeTriBITSCompatibility
+  SOURCES UnitTest_CMakeTriBITSCompatibility.cpp
+)
+
+SET(Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityDisable_DISABLE ON)
+KOKKOS_ADD_TEST(
+  NAME CoreUnitTest_CMakeTriBITSCompatibilityDisable
+  EXE CoreUnitTest_CMakeTriBITSCompatibility
+)
+
+SET(Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityExtraArgs_EXTRA_ARGS "--kokkos-test-tribits-compatibility=1")
+KOKKOS_ADD_TEST(
+  NAME CoreUnitTest_CMakeTriBITSCompatibilityExtraArgs
+  EXE CoreUnitTest_CMakeTriBITSCompatibility
+)
+
+SET(Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityEnvironment_ENVIRONMENT "KOKKOS_TEST_TRIBITS_COMPATIBILITY=1")
+KOKKOS_ADD_TEST(
+  NAME CoreUnitTest_CMakeTriBITSCompatibilityEnvironment
+  EXE CoreUnitTest_CMakeTriBITSCompatibility
+)
```

Comment:
Setting `ENVIRONMENT` via a CMake variable is not supported by `TriBITS` but only by `tribits_add_test` directly. Do we want to keep it? I now lean slightly to removing until there is demand.

Also, we could add `<fullTestName>_SET_DISABLED_AND_MSG` for better compatibility with `TriBITS`.

Reply:
I know it is not a TriBITS option but could not think of a better way to categorize it.
I could add a comment if you want me to.

As far as _SET_DISABLED_AND_MSG is concerned, I was not planning to tackle it here.